### ISSUE TITLE
delete edge resulted in NULL recode and graphdb structure corrupt.

### ISSUE THREAD

**Issue by hazzasym** [orientechnologies/orientdb#7164]
### OrientDB Version: 
community v2.2.15 (build 2.2.x@rebf7b80c08613e871633edeb39c5c985e26b176c; 2017-01-19 07:39:00+0000) 

### Java Version:  
openjdk version "1.8.0_112"
OpenJDK Runtime Environment (build 1.8.0_112-b16)
OpenJDK 64-Bit Server VM (build 25.112-b16, mixed mode) 

## Expected behavior  
Delete edge  from two vertices  should delete edge correctly and graphdb structure should not corrupted.

## Actual behavior  
When sending command to delete edge from two vertices at the same time ( our application is multi thread and concurrency approach) graph database sometimes result with corrupted database and we should login to console and using command "repair database --fix-graph" to repair.

## Steps to reproduce for example
begin
create edge from vertexA to vertexB. (assume we get edge E1)
create edge from vertexA to vertexC. (assume we get edge E2)
delete edge from vertexA to ( select statement that return vertexB and vertexC )
commit.

then send above command to thread-1 and thread-2 at the same time.

sometimes results from these step if we query vertexA we still get E1 or E2 show in record, but actually these edges are disappeared, and we get null if we try to load them or query them.

Below is Orientdb error log showing to us when we try to query from vertexA, which has broken link.

2017-02-09 15:32:01:538 SEVER Internal server error:
com.orientechnologies.orient.core.exception.OCommandExecutionException: expression item 'type' cannot be resolved because current record is NULL
        DB name="FTO-Structure-2.0" [ONetworkProtocolHttpDb]

for more information  
Our application use javascript command to send command to OrientDB.
Ex. 

```
var graph = orient.getGraph();
...
...
...
var id = "#14:1658";
....
....
graph.command(new com.orientechnologies.orient.core.command.script.OCommandScript("sql","delete edge from " + id + " to (select from items where name in ['inbox',
'share'] and owner in " + userslist + " )")).execute();
......
graph.commit()
```

So, another question is if we use this graph command with "orient.getGraph()" this should create transactional operation to database for us on the fly or not?
If yes, why our application sending this script at the same time to OrientDB, the database should return failed or at least not corrupt? Please correct me if I missed understand some point and OrientDB usage.

If you have any questions or need more informations don't hesitate to contact me.
Thank you for great product.

Thanks,

**Comment by hazzasym** [comment]
After waiting for a long time. I notice that this issue maybe relate to #7118?

**Comment by tglman** [comment]
hi @hazzasym,

Yes we had an issue on delete with graph in 2.2.15 and we fixed in in 2.2.16.

Regards

**Comment by hazzasym** [comment]
Sure, I will close this issue and then check this problem again with latest release.

Best Regards,
